src/hydra/dataProvider.js

```diff
@@ -564,5 +568,53 @@ export default (
                   (status ? `Status: ${status}` : ''),
               );
             }),
+    subscribe: (resourceIDs, callback) => {
+      resourceIDs.forEach((resourceID) => {
+        const sub = subscriptions.find((sub) => sub.id === resourceID);
+        if (sub != null) {
+          sub.count++;
+          return;
+        }
+
+        const url = new URL(mercureHub, window.origin);
+        url.searchParams.append(
+          'topic',
+          new URL(resourceID, entrypoint).toString(),
+        );
+        const eventSource = new EventSource(url.toString());
+        const eventListener = (event) => {
+          const document = transformJsonLdDocumentToReactAdminDocument(
+            JSON.parse(event.data),
+          );
+          // we need redux's `dispatch` from the react tree
```

Comment:
I don't understand this comment.

Reply:
The callback to the hook is necessary only because we need to dispatch an action, if we could dispatch from the `dataProvider`, we could do everything from withing `subscribe`.

I tried to word it better